test/DynamoCoreWpfTests/PreferencesViewModelTests.cs

```diff
@@ -20,10 +24,59 @@ public void SelectedPackagePathForInstall_Setter()
             // This setter also affects the default package directory of the PackageLoader.
             preferencesVM.SelectedPackagePathForInstall = Path.GetTempPath();
 
+            // Simulate exiting preference dialog
+            preferencesVM.CommitPackagePathsForInstall();
+
             var pathManager = ViewModel.Model.PathManager;
             Assert.AreEqual(Path.GetTempPath(), pathManager.DefaultPackagesDirectory);
         }
 
+        [Test]
+        public void PackagePathForInstall_Add_Update_Remove_Paths()
+        {
+            var preferencesVM = ViewModel.PreferencesViewModel;
+            Assert.NotNull(preferencesVM);
+
+            // Simulate opening preference dialog
+            preferencesVM.PackagePathsViewModel.InitializeRootLocations();
+            preferencesVM.InitPackagePathsForInstall();
+
+            // The default selected package path for download is AppData.
+            var selectedPath = preferencesVM.SelectedPackagePathForInstall;
+            Assert.AreEqual(GetAppDataFolder(), selectedPath);
+
+            // Add a new path
+            preferencesVM.PackagePathsViewModel.RootLocations.Add(@"C:\");
+
+            // Check that the path is added to the list of paths for install
+            Assert.AreEqual(@"C:\", preferencesVM.PackagePathsForInstall.Last());
+
+            // Update path
+            preferencesVM.SelectedPackagePathForInstall = @"C:\";
+            preferencesVM.PackagePathsViewModel.RootLocations[
+                preferencesVM.PackagePathsViewModel.RootLocations.IndexOf(@"C:\")] = @"D:\";
+
+            // Check that directory has been updated in the list of paths for install
+            Assert.AreEqual(@"D:\", preferencesVM.PackagePathsForInstall.Last());
+
+            // Check that the selection has updated
+            Assert.AreEqual(@"D:\", preferencesVM.SelectedPackagePathForInstall);
+
+            // Remove path
+            preferencesVM.PackagePathsViewModel.RootLocations.Remove(@"D:\");
+
+            // Check that the selection has changed and is not empty
+            var selection = preferencesVM.SelectedPackagePathForInstall;
+            Assert.AreNotEqual(@"D:\", selection);
+            Assert.IsFalse(string.IsNullOrEmpty(selection));
```

Comment:
What is `selection` in this case - is it appdata?

Reply:
yes

Reply:
Can we add an assertion verifying that that is the case?

Reply:
Actually, since this is a cherry-pick, never mind. Minor point.

Reply:
Sure